### PR TITLE
[Feature] adapt limit for sqlserver (#48214)

### LINKED ISSUE #48214: Adapt limit for sqlserver
- [x] https://github.com/StarRocks/starrocks/pull/48248

 while sqlserver use top(xx) as key work for limit, sr didn't adapt this.

you can verify by `select * from sqlservertable limit 1;`

the code is about jdbc_connector.cpp:get_jdbc_sql()

## Feature request

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by XinzhuangL**
## What I'm doing:

Fixes #48214

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/connector/jdbc_connector.cpp [hunk 1]
```diff
@@ -51,6 +51,10 @@ static std::string get_jdbc_sql(const Slice jdbc_url, const std::string& table,
                                 const std::vector<std::string>& filters, int64_t limit) {
     std::ostringstream oss;
     oss << "SELECT";
+    if (limit != -1 && jdbc_url.starts_with("jdbc:sqlserver")) {
```
Comment (zombee0) [inline be/src/connector/jdbc_connector.cpp:54]: i think for sqlserver the sql will be select top(xx) xxx from xxx limit xxx;

Comment (XinzhuangL) [reply be/src/connector/jdbc_connector.cpp:54]: i set the limit=-1, this code  "if (limit != -1) {
        if (jdbc_url.starts_with("jdbc:oracle")) {
            // oracle doesn't support limit clause, we should generate a subquery to do this
            // ref: https://stackoverflow.com/questions/470542/how-do-i-limit-the-number-of-rows-returned-by-an-oracle-query-after-ordering
            return fmt::format("SELECT * FROM ({}) WHERE ROWNUM <= {}", oss.str(), limit);
        } else {
            oss << " LIMIT " << limit;
        }
    }"   is no longer valid ?

Comment (zombee0) [reply be/src/connector/jdbc_connector.cpp:54]: oh, got it.

**Review by zombee0 — APPROVED** [review]

**Comment by zombee0** [comment]
https://github.com/Mergifyio backport branch-3.3

**Pull request merged**